packages/frontend/src/pages/app/index.tsx

```diff
@@ -29,19 +34,28 @@ const DEFAULT_LOCATION = {
 };
 
 interface AppProps {
-    customBaseURL?: string;
+    kpiTokenTemplateBaseURL?: string;
+    oracleTemplateBaseURL?: string;
     templateId?: number;
 }
 
-export const App = ({ customBaseURL, templateId }: AppProps) => {
+export const App = ({
+    kpiTokenTemplateBaseURL,
+    oracleTemplateBaseURL,
+    templateId,
+}: AppProps) => {
     const location = useLocation();
     const previousLocation = usePreviousDistinct(location);
     const navigate = useNavigate();
-    const { setPreferDecentralization } = usePreferencesSetters();
+    const setPreferDecentralization = useSetPreferDecentralization();
+    const setDevMode = useSetDevMode();
+    const setKPITokenTemplateBaseURL = useSetKPITokenTemplateBaseURL();
+    const setOracleTemplateBaseURL = useSetOracleTemplateBaseURL();
 
-    useEffect(() => {
-        if (__PREVIEW_MODE__) setPreferDecentralization(true);
-    }, [setPreferDecentralization]);
+    if (__PREVIEW_MODE__) setPreferDecentralization(true);
```

Comment:
if `__PREVIEW_MODE__` is a boolean we can do it like bellow.

```suggestion
   setPreferDecentralization(__PREVIEW_MODE__);
```

Reply:
`__PREVIEW_MODE__` is a global defined by both Webpack and Rollup (with different values) at build time. Since the substitution is done before the optimization step it's more efficient to do it like this because when `__PREVIEW_MODE__` is false `if (__PREVIEW_MODE__) setPreferDecentralization(true);` gets rewritten to `if (false) setPreferDecentralization(true);`, which results in the whole instruction being excluded from the final bundle.